File: algoexpert/java/Sorted_Squared_Array.java
import java.util.*;

class Program {

  public int[] sortedSquaredArray(int[] array) {
    int n = array.length, left = 0, right = n - 1;
     
    int result[] = new int[n];
     
    for(int index = n - 1; index >= 0; index--)
    {
        if (Math.abs(array[left]) > array[right])
        {
            result[index] = array[left] * array[left];
            left++;
        }
        else
        {
            result[index] = array[right] * array[right];
            right--;
        }
    }
    return result;
  }
}

File: algoexpert/python/Sorted_Squared_Array.py
def sortedSquaredArray(array):
    n = len(array)
    left = 0
    right = n - 1
    result = [0] * n

    for index in range(n - 1, -1, -1):
        if abs(array[left]) > array[right]:
            result[index] = array[left] * array[left]
            left += 1
        else:
            result[index] = array[right] * array[right]
            right -= 1

    return result